### DISCUSSION TITLE
Face recognition error : not enough memory allocated

### DISCUSSION THREAD

**Discussion by Dams07** [nextcloud/all-in-one#4894] [Questions] [answered]
Hi,

I use facerecognition community container and i run into an issue with my background_job since a few days : 

```
1/8 - Executing task CheckRequirementsTask (Check all requirements)
2/8 - Executing task CheckCronTask (Check that service is started from either cron or from command)
3/8 - Executing task DisabledUserRemovalTask (Purge all the information of a user when disable the analysis.)
4/8 - Executing task StaleImagesRemovalTask (Crawl for stale images (either missing in filesystem or under .nomedia) and remove them from DB)
5/8 - Executing task CreateClustersTask (Create new persons or update existing persons)
        Face clustering will be recreated with new information or changes
        There are 47580 faces for clustering
PHP Fatal error:  Allowed memory size of 2147483648 bytes exhausted (tried to allocate 134217736 bytes) in /var/www/html/custom_apps/facerecognition/lib/Clusterer/ChineseWhispers.php on line 156
``` 

I think it appears since this commit (it decrease amount of memory allowed) : [here](https://github.com/nextcloud/all-in-one/commit/ed339d5101ece3fef6cc7ba7f86e8bf614128e48) ?

Is there a way to customize these values ?
My server have 64GB ram, so memory is not an issue for me.

Thank you very much for your help 👍

**Comment by szaimen** [comment]
CC @matiasdelellis

**Comment by matiasdelellis** [comment, marked as answer]
Hi @Dams07
You can try the batching option. It wasn't intended for this, but it seems to help a lot for it.

https://github.com/matiasdelellis/facerecognition/wiki/Batched-face-clustering

Regarding the memory limit, I could increase it again, (I don't use AIO but I understand that you can increase it manually), however clustering times are still a problem. Therefore, if it is enough, now I prefer to use batches, and focus on improving memory consumption.

**Comment by Dams07** [comment]
Hi,

It seems to work, I don't have error anymore.
Thank you.

@matiasdelellis : I have another question, how to change the FACE_MODEL number ?
It seems to be forced to 3 in facerecognition.json.

@szaimen : I think, it is missing a feature in AIO to pass somes parameters to community containers in order to modify their behavior.
